vyper/function_signature.py

```diff
@@ -84,7 +84,9 @@ def from_definition(cls, code, _sigs=None):
             else:
                 raise StructureException("Bad decorator", dec)
         if public and private:
-            raise StructureException("Cannot use public and private decorators on the same function", code)
+            raise StructureException("Cannot use public and private decorators on the same function: {}".format(name))
+        if payable and const:
+            raise StructureException("Function `{}` cannot have a constant function that's payable.".format(name))
```

Comment:
"Function `{}` cannot have a constant function that's payable." should be "Function `{}` cannot be both constant and payable." for clarities' sake